posthog/tasks/alerts/checks.py

```diff
@@ -1,23 +1,29 @@
+from typing import Optional
+
 from celery import shared_task
 from celery.canvas import group, chain
+from django.db import transaction
 from django.utils import timezone
 import math
 import structlog
+from sentry_sdk import capture_exception
 
 from posthog.api.services.query import ExecutionMode
 from posthog.caching.calculate_results import calculate_for_query_based_insight
 from posthog.email import EmailMessage
+from posthog.errors import CHQueryErrorTooManySimultaneousQueries
 from posthog.hogql_queries.legacy_compatibility.flagged_conversion_manager import (
     conversion_to_query_based,
 )
-from posthog.models import Alert
-from posthog.schema import AnomalyCondition
+from posthog.models import AlertConfiguration
+from posthog.models.alert import AlertCheck
+from posthog.tasks.utils import CeleryQueue
 
 logger = structlog.get_logger(__name__)
 
 
 def check_all_alerts() -> None:
```

Comment:
Yeah, I have to think about it... in terms of limiting workload for query calculations we might have the insights with checks also calculate during warming, and then we rely on cached results as far as possible. (Goal is currently not real time alerts here, so that would be fine). For checking alerts though we might also put them on a limit worker that doesn't scale up much. And then we might just chain the alerts within one team and configure expirations nicely, like we do with warming.